src/main/modules/ardoq/ArdoqClient.ts

```diff
@@ -0,0 +1,93 @@
+import { ArdoqComponentCreatedResponse } from './ArdoqComponentCreatedResponse';
+import { Dependency } from './Dependency';
+
+import axios from 'axios';
+
+export class ArdoqClient {
+  apiKey: string;
+  apiUrl: string;
+  apiWorkspace: string;
+  cache: Map<string, string> = new Map<string, string>();
+
+  constructor(apiKey: string, apiUrl: string, apiWorkspace: string) {
+    this.apiKey = apiKey;
+    this.apiUrl = apiUrl;
+    this.apiWorkspace = apiWorkspace;
+  }
+
+  private cacheResult(d: Dependency) {
+    this.cache.set(d.name, d.version);
+  }
+
+  private isCached(d: Dependency): boolean {
+    return this.cache.get(d.name) === d.version;
+  }
+
+  private searchForComponent(dependencyFullName: string) {
+    const url = this.apiUrl + '/api/component/search';
+    return axios.get(url, {
+      params: {
+        workspace: this.apiWorkspace,
+        name: dependencyFullName,
+      },
+      headers: {
+        Authorization: 'Token token=' + this.apiKey,
+      },
+      responseType: 'json',
+    });
+  }
+
+  private createComponent(dependencyFullName: string) {
+    const url = this.apiUrl + '/api/component/search';
+    return axios.post(
+      url,
+      {
+        rootWorkspace: this.apiWorkspace,
+        name: dependencyFullName,
+      },
+      {
+        params: {
+          workspace: this.apiWorkspace,
+          name: dependencyFullName,
+        },
+        headers: {
+          Authorization: 'Token token=' + this.apiKey,
+        },
+        responseType: 'json',
+      }
+    );
+  }
+
+  async updateDep(d: Dependency): Promise<ArdoqComponentCreatedResponse> {
+    if (this.isCached(d)) {
+      return ArdoqComponentCreatedResponse.EXISTING;
+    }
+
+    return this.searchForComponent(d.getFullName()).then(searchResponse => {
```

Comment:
rather than using `.then` use `const searchResponse = this.searchForComponent(...);`